kore/src/Kore/Log/DebugTransition.hs

```diff
@@ -0,0 +1,117 @@
+{- |
+Copyright   : (c) Runtime Verification, 2020
+License     : NCSA
+-}
+module Kore.Log.DebugTransition (
+    DebugTransition (..),
+    debugBeforeTransition,
+    debugAfterTransition,
+    debugFinalTransition,
+) where
+
+import Kore.Attribute.Axiom (
+    SourceLocation,
+ )
+import Kore.Reachability.ClaimState (
+    ClaimState,
+ )
+import Kore.Reachability.Prim (
+    Prim (..),
+ )
+import Kore.Reachability.SomeClaim (
+    SomeClaim (..),
+ )
+import Log
+import Prelude.Kore
+import Pretty (
+    Pretty (..),
+ )
+import qualified Pretty
+
+data DebugTransition
+    = DebugAfterTransition
+        (Maybe (ClaimState SomeClaim))
+        Prim
+        [SourceLocation]
+    | DebugBeforeTransition
+        (ClaimState SomeClaim)
+        Prim
+    deriving stock (Show)
+
+instance Pretty DebugTransition where
+    pretty (DebugAfterTransition result transition appliedRules) =
+        Pretty.vsep $
+            concat
+                [
+                    [ "Applied the following transition:"
+                    , Pretty.indent 4 (pretty transition)
+                    ]
+                , prettyRules
```

Comment:
Any information about rules should only be printed if the transition itself is one which applies rules (`ApplyClaims` or `ApplyAxioms`).